### PR TITLE
DTD support with xml.fileAssociations

### PR DESCRIPTION
**Pull request by angelozerr**
DTD support with xml.fileAssociations

See https://github.com/redhat-developer/vscode-xml/issues/184

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
To test this PR, use XML file association like explained https://github.com/redhat-developer/vscode-xml/wiki/XMLValidation#xml-file-association but with the DTD:

```xml
<!ELEMENT foo (bar)>
<!ELEMENT bar EMPTY>
```

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/settings/XMLFileAssociation.java [hunk 1]
```diff
@@ -39,7 +40,13 @@ public void setSystemId(String systemId) {
 	public Map<String, String> getExternalSchemaLocation() {
 		if (externalSchemaLocation == null) {
 			this.externalSchemaLocation = new HashMap<String, String>();
-			this.externalSchemaLocation.put(IExternalSchemaLocationProvider.NO_NAMESPACE_SCHEMA_LOCATION, systemId);
+			if (DOMUtils.isXSD(systemId)) {
+				// It's a XSD
```
Comment (datho7561) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/settings/XMLFileAssociation.java:44]: Probably don't need this comment

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/settings/XMLFileAssociation.java [hunk 2]
```diff
@@ -39,7 +40,13 @@ public void setSystemId(String systemId) {
 	public Map<String, String> getExternalSchemaLocation() {
 		if (externalSchemaLocation == null) {
 			this.externalSchemaLocation = new HashMap<String, String>();
-			this.externalSchemaLocation.put(IExternalSchemaLocationProvider.NO_NAMESPACE_SCHEMA_LOCATION, systemId);
+			if (DOMUtils.isXSD(systemId)) {
+				// It's a XSD
+				this.externalSchemaLocation.put(IExternalGrammarLocationProvider.NO_NAMESPACE_SCHEMA_LOCATION, systemId);
+			} else {
+				// It's a DTD
```
Comment (datho7561) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/settings/XMLFileAssociation.java:47]: Assume it's a DTD

Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/settings/XMLFileAssociation.java]: fixed

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/settings/XMLFileAssociation.java (hunk 1 above)
Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/settings/XMLFileAssociation.java]: fixed

**Pull request merged**